client/src/main/java/org/eclipse/hono/client/NoConsumerException.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client;
+
+import java.net.HttpURLConnection;
+
+/**
+ * A {@link ServerErrorException} indicating that there is no consumer for the request message.
+ */
+public class NoConsumerException extends ServerErrorException {
+
+    /**
+     * Resource key for the client facing error message.
+     */
+    public static final String CLIENT_FACING_MESSAGE_KEY = "SERVER_ERROR_NO_CONSUMER";
+
+    /**
+     * Creates a new NoConsumerException.
```

Comment:
IMHO it would be helpful to explain that apart from the given message, the *clientFacingMessage* will be set to a certain key. This is also true for the other exception classes. WDYT?

Reply:
I've amended the commit accordingly.